[Bug]: netdata build fails if `-DENABLE_DBENGINE:BOOL=FALSE` set

### Bug description

```console
...
FAILED: [code=1] CMakeFiles/netdata.dir/src/web/api/queries/weights.c.o
/nix/store/ag7fc58qzff4j1sgd7v98irhcr6vcg8v-gcc-wrapper-14.3.0/bin/gcc -DPROTOBUF_USE_DLLS -D_GNU_SOURCE -I/build/source/build/src/aclk/aclk-schemas -I/nix/store/r2f426l03i48gqi8ycjs3dp3671h6dar-util-linux-minimal-2.41.2-dev/include/uuid >
/build/source/src/web/api/queries/weights.c: In function 'query_scope_foreach_host_parallel':
/build/source/src/web/api/queries/weights.c:2274:9: error: implicit declaration of function 'rrdeng_enq_cmd' [-Wimplicit-function-declaration]
 2274 |         rrdeng_enq_cmd(NULL, RRDENG_OPCODE_PARALLEL_WEIGHT, &thread_data[i], &thread_data[i].completion, STORAGE_PRIORITY_INTERNAL_DBENGINE, NULL, NULL);
      |         ^~~~~~~~~~~~~~
/build/source/src/web/api/queries/weights.c:2274:30: error: 'RRDENG_OPCODE_PARALLEL_WEIGHT' undeclared (first use in this function)
 2274 |         rrdeng_enq_cmd(NULL, RRDENG_OPCODE_PARALLEL_WEIGHT, &thread_data[i], &thread_data[i].completion, STORAGE_PRIORITY_INTERNAL_DBENGINE, NULL, NULL);
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/web/api/queries/weights.c:2274:30: note: each undeclared identifier is reported only once for each function it appears in
...
```

### Expected behavior

Should build with no issue

### Steps to reproduce

1. Build with cmake and use `-DENABLE_DBENGINE:BOOL=FALSE`

### Installation method

other

### System info

```shell
Linux HOSTNAME 6.17.7 #1-NixOS SMP PREEMPT_DYNAMIC Sun Nov  2 13:18:05 UTC 2025 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_CODENAME=xantusia
/etc/lsb-release:DISTRIB_DESCRIPTION="NixOS 25.11 (Xantusia)"
/etc/lsb-release:DISTRIB_ID=nixos
/etc/lsb-release:DISTRIB_RELEASE="25.11"
/etc/lsb-release:LSB_VERSION="25.11 (Xantusia)"
/etc/os-release:ANSI_COLOR="0;38;2;126;186;228"
/etc/os-release:BUILD_ID="25.11.20251117.89c2b23"
/etc/os-release:CPE_NAME="cpe:/o:nixos:nixos:25.11"
/etc/os-release:DEFAULT_HOSTNAME=nixos
/etc/os-release:ID=nixos
/etc/os-release:ID_LIKE=""
/etc/os-release:IMAGE_ID=""
/etc/os-release:IMAGE_VERSION=""
/etc/os-release:LOGO="nix-snowflake"
/etc/os-release:NAME=NixOS
/etc/os-release:PRETTY_NAME="NixOS 25.11 (Xantusia)"
/etc/os-release:VARIANT=""
/etc/os-release:VARIANT_ID=""
/etc/os-release:VENDOR_NAME=NixOS
/etc/os-release:VERSION="25.11 (Xantusia)"
/etc/os-release:VERSION_CODENAME=xantusia
/etc/os-release:VERSION_ID="25.11"
```

### Netdata build info

```shell
The package is not yet built.
```

### Additional info

Tag `v2.8.0`

--- a/src/web/api/queries/weights.c
+++ b/src/web/api/queries/weights.c
@@ -2226,6 +2226,12 @@ static ssize_t weights_do_context_callback(void *data, RRDCONTEXT_ACQUIRED *rca,
 static ssize_t query_scope_foreach_host_parallel(SIMPLE_PATTERN *scope_hosts_sp, SIMPLE_PATTERN *hosts_sp,
                                                   struct query_weights_data *qwd)
 {
+#ifndef ENABLE_DBENGINE
+    return query_scope_foreach_host(scope_hosts_sp, hosts_sp,
+                                    weights_do_node_callback, qwd,
+                                    &qwd->versions, NULL);
+
+#else
     size_t host_count = dictionary_entries(rrdhost_root_index);
     qwd->hosts_array = mallocz(sizeof(RRDHOST *) * host_count);
     qwd->hosts_array_capacity = host_count;
@@ -2305,6 +2311,7 @@ static ssize_t query_scope_foreach_host_parallel(SIMPLE_PATTERN *scope_hosts_sp,
     freez(qwd->hosts_array);
 
     return total_added;
+#endif
 }
 
 static ssize_t weights_do_node_callback(void *data, RRDHOST *host, bool queryable) {

